BarcodeParserBuilder/Barcodes/CODE128/Code128StringParserBuilder.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using BarcodeParserBuilder.Infrastructure;
+
+namespace BarcodeParserBuilder.Barcodes.CODE128
+{
+    internal class Code128StringParserBuilder : BaseFieldParserBuilder<string?>
+    {
+        protected override string? Build(string? obj) => string.IsNullOrWhiteSpace(obj) ? null : obj;
+        protected override string? Parse(string? value) => value;
+
+        protected override bool Validate(string? value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                return true;
+
+            return true;
+        }
+
+        /// <summary>
+        /// The validation currently does _not_ depend from the reader symbology identifier since all Code128 barcodes have same encoding
+        /// It is very unlikely that such old standard will evolve but we keep the method signature
+        /// </summary>
+        /// <param name="value">reading</param>
+        /// <param name="information">AimSymbologyIdentifier for the Code39</param>
+        /// <returns></returns>
+        protected internal static bool Validate(string? value, Code128SymbologyIdentifier information)
+        {
+            return ValidateFullASCII(value);
+        }
+
+        /// <summary>
+        /// Code128 contains full ASCII set of symbols
+        /// </summary>
+        /// <param name="value">reading</param>
+        /// <returns>whether the reading is subset of full ASCII</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        protected internal static bool ValidateFullASCII(string? value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            return System.Text.RegularExpressions.Regex.IsMatch(value!, "^[\x0-\xFF]+$");
```

Comment:
isn't there like a string.all(char.IsAscii); method you could use?

Reply:
According to [MSDN](https://learn.microsoft.com/en-us/dotnet/api/system.char.isascii?view=net-8.0), isnt it that: Per http://www.unicode.org/glossary/#ASCII, ASCII is only U+0000..U+007F.

Reply:
ah, i thought char.IsAscii had a 00 - FF limt. my mistake ^^;
makes sense if you think about it. C#'s char & string are all utf-16 internally.